samples/boards/sensortile_box_pro/sensors-on-board/src/main.c

```diff
@@ -0,0 +1,302 @@
+		printk("%s: device not ready.\n", led1_gpio.port->name);
+		return -1;
+	}
+	gpio_pin_configure_dt(&led1_gpio, GPIO_OUTPUT_INACTIVE);
+
+	/* led 2 */
+	if (!gpio_is_ready_dt(&led2_gpio)) {
+		printk("%s: device not ready.\n", led2_gpio.port->name);
+		return -1;
+	}
+	gpio_pin_configure_dt(&led2_gpio, GPIO_OUTPUT_INACTIVE);
+
+	/* led 3 */
+	if (!gpio_is_ready_dt(&led3_gpio)) {
+		printk("%s: device not ready.\n", led3_gpio.port->name);
+		return -1;
+	}
+	gpio_pin_configure_dt(&led3_gpio, GPIO_OUTPUT_INACTIVE);
+
+	/* output led pattern */
+	for (i = 0; i < 12; i++) {
+		gpio_pin_set_dt(&led0_gpio, ((i % 4) == 0) ? 1 : 0);
+		gpio_pin_set_dt(&led1_gpio, ((i % 4) == 1) ? 1 : 0);
+		gpio_pin_set_dt(&led2_gpio, ((i % 4) == 2) ? 1 : 0);
+		gpio_pin_set_dt(&led3_gpio, ((i % 4) == 3) ? 1 : 0);
+		k_sleep(K_MSEC(100));
+	}
+
+	/* turn all leds off */
+	gpio_pin_set_dt(&led0_gpio, 0);
+	gpio_pin_set_dt(&led1_gpio, 0);
+	gpio_pin_set_dt(&led2_gpio, 0);
+	gpio_pin_set_dt(&led3_gpio, 0);
+
+	return 0;
+}
+
+int main(void)
+{
+	int cnt = 1;
+
+	/* signal that sample is started */
+	if (led_pattern_out() < 0) {
+		return -1;
+	}
+
+	printk("SensorTile.box Pro sensor test\n");
+
+	const struct device *const hts221 = DEVICE_DT_GET_ONE(st_hts221);
+	const struct device *const lsm6dsv16x = DEVICE_DT_GET_ONE(st_lsm6dsv16x);
+	const struct device *const lis2mdl = DEVICE_DT_GET_ONE(st_lis2mdl);
+
+	if (!device_is_ready(hts221)) {
+		printk("%s: device not ready.\n", hts221->name);
+		return 0;
+	}
+	if (!device_is_ready(lsm6dsv16x)) {
+		printk("%s: device not ready.\n", lsm6dsv16x->name);
+		return 0;
+	}
+	if (!device_is_ready(lis2mdl)) {
+		printk("%s: device not ready.\n", lis2mdl->name);
+		return 0;
+	}
+
+	lis2mdl_config(lis2mdl);
+	lsm6dsv16x_config(lsm6dsv16x);
+
+	while (1) {
+		struct sensor_value hts221_hum, hts221_temp;
+		struct sensor_value lsm6dsv16x_accel[3], lsm6dsv16x_gyro[3];
+		struct sensor_value lis2mdl_magn[3];
+		struct sensor_value lis2mdl_temp;
+
+		/* handle HTS221 sensor */
+		if (sensor_sample_fetch(hts221) < 0) {
+			printf("HTS221 Sensor sample update error\n");
+			return 0;
+		}
+
+#ifndef CONFIG_LSM6DSV16X_TRIGGER
+		if (sensor_sample_fetch(lsm6dsv16x) < 0) {
+			printf("LSM6DSV16X Sensor sample update error\n");
+			return 0;
+		}
+#endif
+
+#ifndef CONFIG_LIS2MDL_TRIGGER
+		if (sensor_sample_fetch(lis2mdl) < 0) {
+			printf("LIS2MDL Magn Sensor sample update error\n");
+			return 0;
+		}
+#endif
+
+		sensor_channel_get(hts221, SENSOR_CHAN_HUMIDITY, &hts221_hum);
+		sensor_channel_get(hts221, SENSOR_CHAN_AMBIENT_TEMP, &hts221_temp);
+		sensor_channel_get(lsm6dsv16x, SENSOR_CHAN_ACCEL_XYZ, lsm6dsv16x_accel);
+		sensor_channel_get(lsm6dsv16x, SENSOR_CHAN_GYRO_XYZ, lsm6dsv16x_gyro);
+		sensor_channel_get(lis2mdl, SENSOR_CHAN_MAGN_XYZ, lis2mdl_magn);
+		sensor_channel_get(lis2mdl, SENSOR_CHAN_DIE_TEMP, &lis2mdl_temp);
+
+		/* Display sensor data */
+
+		/* Erase previous */
+		printf("\0033\014");
```

Comment:
wow, is this a like a clear terminal? may be worth expanding the comment a bit

Reply:
yes, it is.
OK, I will add some additional words to the comment. It makes sense